piglot/solver/abaqus/reader.py

```diff
@@ -200,15 +200,100 @@ def write_output_file(i, variables_array, variable, step, location, file_name):
+    variables : dict
+        A dictionary containing the step name under the key "step_name".
+
+    Returns
+    -------
+    step : Step
+        The step from the output database that matches the name provided in the variables 
+        dictionary.
+
+    Raises
+    ------
+    ValueError
+        Raises an error if the step name is not found in the output database.
+    """
+    step_names_list = list(odb.steps.keys())
+    step_names_list_upper = [stepName.upper() for stepName in step_names_list]
+    step_name_upper = variables["step_name"].upper()
+    if step_name_upper not in step_names_list_upper:
+        raise ValueError("Step name not found in the output database.")
+    step_name = step_names_list[step_names_list_upper.index(step_name_upper)]
+    return odb.steps[step_name]
+
+def get_instance(odb, variables):
+    """Create a variable that refers to the respective instance.
+
+    Parameters
+    ----------
+    odb : Odb
+        An instance of the Odb class from the Abaqus scripting interface, representing the output 
+        database.
+    variables : dict
+        A dictionary containing the instance name under the key "instance_name".
+
+    Returns
+    -------
+    instance : str
+        The instance from the output database that matches the name provided in the variables 
+        dictionary.
+
+    Raises
+    ------
+    ValueError
+        Raises an error if the instance name is not found in the output database.
+    """
+    instance_names_list = list(odb.rootAssembly.instances.keys())
+    instance_names_list_upper = [instanceName.upper() for instanceName in instance_names_list]
+    instance_name_upper = variables["instance_name"].upper()
+    if instance_name_upper not in instance_names_list_upper:
+        raise ValueError("Instance name not found in the output database.")
+
+    return instance_names_list[instance_names_list_upper.index(instance_name_upper)]
+
+def get_set(odb, variables, instance):
+    """Create a variable that refers to the respective set.
+
+    Parameters
+    ----------
+    odb : Odb
+        An instance of the Odb class from the Abaqus scripting interface, representing the output 
+        database.
+    variables : dict
+        A dictionary containing the set name under the key "set_name".
+
+    Returns
+    -------
+    set : str
+        The set from the output database that matches the name provided in the variables 
+        dictionary.
+
+    Raises
+    ------
+    ValueError
+        Raises an error if the set name is not found in the output database.
+    """
+    set_names_list = list(odb.rootAssembly.instances[instance].nodeSets.keys())
+    set_names_list_upper = [setName.upper() for setName in set_names_list]
+    set_name_upper = variables["set_name"].upper()
+    if set_name_upper not in set_names_list_upper:
+        raise ValueError("Set name not found in the output database.")
+    return set_names_list[set_names_list_upper.index(set_name_upper)]
+
```

Comment:
This block seems repeating. Consider generalising the case-insensitive searching for a key in a list of strings to a separate function, and then using it to get the step, instance and nodeset names.

Reply:
Done. Thank you for the suggestion.